client_mining_p/blockchain.py

```diff
@@ -1,2 +1,176 @@
+        :param block": <dict> Block
+        "return": <str>
+        """
+        # Use json.dumps to convert json into a string
+        # Use hashlib.sha256 to create a hash
+        # It requires a `bytes-like` object, which is what
+        # .encode() does.
+        # It convertes the string to bytes.
+        # We must make sure that the Dictionary is Ordered,
+        # or we'll have inconsistent hashes
+        # TODO: Create the block_string
+        string_object = json.dumps(block, sort_keys=True)
+        block_string = string_object.encode()
+        # TODO: Hash this string using sha256
+        raw_hash = hashlib.sha256(block_string)
+        hex_hash = raw_hash.hexdigest()
+        # By itself, the sha256 function returns the hash in a raw string
+        # that will likely include escaped characters.
+        # This can be hard to read, but .hexdigest() converts the
+        # hash to a string of hexadecimal characters, which is
+        # easier to work with and understand
+        # TODO: Return the hashed block string in hexadecimal format
+        return hex_hash
+    @property
+    def last_block(self):
+        return self.chain[-1]
+    # def proof_of_work(self, block):
+    #     """
+    #     Simple Proof of Work Algorithm
+    #     Stringify the block and look for a proof.
+    #     Loop through possibilities, checking each one against `valid_proof`
+    #     in an effort to find a number that is a valid proof
+    #     :return: A valid proof for the provided block
+    #     """
+    #     # Proof is a SHA256 hash with 3 leading zeroes
+    #     block_string = json.dumps(block, sort_keys=True).encode()
+    #     proof = 0
+    #     while not self.valid_proof(block_string, proof):
+    #         proof += 1
+    #     guess = f'{block_string}{proof}'.encode()
+    #     guess_hash = hashlib.sha256(guess).hexdigest()
+    #     return proof
+    @staticmethod
+    def valid_proof(block_string, proof):
+        """
+        Validates the Proof:  Does hash(block_string, proof) contain 3
+        leading zeroes?  Return true if the proof is valid
+        :param block_string: <string> The stringified block to use to
+        check in combination with `proof`
+        :param proof: <int?> The value that when combined with the
+        stringified previous block results in a hash that has the
+        correct number of leading zeroes.
+        :return: True if the resulting hash is a valid proof, False otherwise
+        """
+        guess = f'{block_string}{proof}'.encode()
+        guess_hash = hashlib.sha256(guess).hexdigest()
+        # return True or False
+        return guess_hash[:6] == "000000"
+# Instantiate our Node
+app = Flask(__name__)
+# Generate a globally unique address for this node
+node_identifier = str(uuid4()).replace('-', '')
+# Instantiate the Blockchain
+blockchain = Blockchain()
+@app.route('/mine', methods=['GET'])
+def mine():
+    # Run the proof of work algorithm to get the next proof
```

Comment:
Awesome! The only suggestion I could give would be to have some logic to handle non-JSON data responses. Other than that, looks good!